cylc/flow/network/resolvers.py

```diff
@@ -87,10 +87,10 @@ def node_filter(node, args):
     """Filter nodes based on attribute arguments"""
     natts = collate_node_atts(node)
     return (
-        (args.get('ghosts') or node.state != '') and
-        (not (args.get('states') and node.state != '')
+        (args.get('ghosts') or natts[5] != '') and
+        (not (args.get('states') and natts[5] != '')
          or node.state in args['states']) and
-        not (args.get('exstates') and node.state != '' and
+        not (args.get('exstates') and natts[5] != '' and
```

Comment:
Maybe add a comment above to explain what the elements in `natts` are?

Reply:
Done. Hope I it's not too verbose (or not verbose enough).